### DISCUSSION TITLE
Is there a reason the weigher only supports u32, while max capacity is u64?

### DISCUSSION THREAD

**Discussion by GalLalouche** [moka-rs/moka#409] [Q&A] [answered]
Basically the title. I'm trying to use moka with size-based-evictions, as detailed in https://docs.rs/moka/latest/moka/future/struct.Cache.html#example-size-based-eviction, but the limitation on u32 entry weights makes this impractical for some use-cases.

**Comment by tatsuya6502** [comment, marked as answer]
Thank you for using moka.

It uses `u32` for entry weight rather than `u64` to reduce memory overhead. A cache instance must hold the weight values of all entries so a small data type (`u32` 4 bytes) is preferable than a large data type (`u64` 8 bytes). It uses `u64` for the max capacity and `weighted_size` (the sum of the weight values of all entries) to avoid arithmetic overflow (e.g. `u32::MAX + 1u32`) . The memory overhead are small for them because a cache instance only needs to hold one `u64` for each. 

Entry weights are relative values so you can scale your original value to fit in the 2^32 range, and scale the max capacity as well.
